src/main/java/io/odpf/firehose/filter/JsonFilter.java

```diff
@@ -0,0 +1,168 @@
+    private final JsonFormat.Printer jsonPrinter;
+
+    /**
+     * Instantiates a new Json filter.
+     *
+     * @param filterConfig    the consumer config
+     * @param instrumentation the instrumentation
+     */
+    public JsonFilter(FilterConfig filterConfig, Instrumentation instrumentation) {
+        objectMapper = new ObjectMapper();
+        JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
+        jsonPrinter = JsonFormat.printer().preservingProtoFieldNames();
+        messageType = filterConfig.getFilterMessageFormat();
+        filterDataSourceType = filterConfig.getFilterJsonDataSource();
+        String protoSchemaClass = filterConfig.getFilterJsonSchemaProtoClass();
+        String filterJsonSchema = filterConfig.getFilterJsonSchema();
+        this.instrumentation = instrumentation;
+        if (filterDataSourceType != NONE) {
+            try {
+                schema = schemaFactory.getSchema(filterJsonSchema);
+            } catch (Exception e) {
+                throw new FilterException("Failed to parse JSON Schema " + e.getMessage());
+            }
+        }
+        if (messageType == PROTOBUF) {
+            try {
+                protoParser = MethodUtils.getAccessibleMethod(Class.forName(protoSchemaClass), "parseFrom", byte[].class);
+            } catch (ClassNotFoundException e) {
+                throw new FilterException("Failed to load Proto schema class " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Log configs.
+     *
+     * @param filterConfig    the filter config
+     * @param instrumentation the instrumentation
+     */
+    public static void logConfigs(FilterConfig filterConfig, Instrumentation instrumentation) {
+        instrumentation.logInfo("\n\tFilter type: {}", filterConfig.getFilterJsonDataSource());
+        if (filterConfig.getFilterJsonDataSource() != NONE) {
+            instrumentation.logInfo("\n\tFilter JSON Schema: {}", filterConfig.getFilterJsonSchema());
+            instrumentation.logInfo("\n\tFilter message type: {}", filterConfig.getFilterMessageFormat());
+            if (filterConfig.getFilterMessageFormat() == PROTOBUF) {
+                instrumentation.logInfo("\n\tMessage Proto class: {}", filterConfig.getFilterJsonSchemaProtoClass());
+            }
+        } else {
+            instrumentation.logInfo("No filter is selected");
+        }
+    }
+
+    /**
+     * method to filter the EsbMessages.
+     *
+     * @param messages the json/protobuf records in binary format that are wrapped in {@link Message}
+     * @return the list of filtered Messages
+     * @throws FilterException the filter exception
+     */
+    @Override
+    public List<Message> filter(List<Message> messages) {
+        if (filterDataSourceType == NONE) {
+            return messages;
+        }
+        List<Message> filteredMessages = new ArrayList<>();
+        for (Message message : messages) {
```

Comment:
```
        messages.stream().filter(message -> {
            byte[] data = (filterDataSourceType.equals(KEY)) ? message.getLogKey() : message.getLogMessage();
            return evaluate(deserialize(data));
        });
        ```

Reply:
fixed

Reply:
cant be done cause `deserialize(data) `throws FilterException which isnt thrown by the functional interface method